Source/ImGui/Private/ImGuiInputState.cpp

```diff
@@ -14,87 +14,116 @@ FImGuiInputState::FImGuiInputState()
 
 void FImGuiInputState::AddCharacter(TCHAR Char)
 {
-	InputCharacters.Add(Char);
+	IO.AddInputCharacter(ImGuiInterops::CastInputChar(Char));
 }
 
-void FImGuiInputState::SetKeyDown(uint32 KeyIndex, bool bIsDown)
+void FImGuiInputState::SetKeyDown(const FKeyEvent& KeyEvent, bool bIsDown)
 {
-	if (KeyIndex < Utilities::GetArraySize(KeysDown))
-	{
-		if (KeysDown[KeyIndex] != bIsDown)
-		{
-			KeysDown[KeyIndex] = bIsDown;
-			KeysUpdateRange.AddPosition(KeyIndex);
-		}
-	}
+	const FKey& Key = KeyEvent.GetKey();
+	SetKeyDown(Key, bIsDown);
 }
 
-void FImGuiInputState::SetMouseDown(uint32 MouseIndex, bool bIsDown)
+void FImGuiInputState::SetKeyDown(const FKey& Key, bool bIsDown)
 {
-	if (MouseIndex < Utilities::GetArraySize(MouseButtonsDown))
+	const ImGuiKey& ImKey = ImGuiInterops::UnrealToImGuiKey(Key);
+	IO.AddKeyEvent(ImKey, bIsDown);
```

Comment:
The UnrealToImGuiKey has `ImGuiKey_LeftShift`, `ImGuiKey_RightShift`, `ImGuiKey_LeftControl`, `ImGuiKey_RightControl`, `ImGuiKey_LeftAlt`, `ImGuiKey_RightAlt`, `ImGuiKey_LeftSuper` and `ImGuiKey_RightSuper` But we also need to pass events for `ImGuiKey_ModShift`, `ImGuiKey_ModControl`, `ImGuiKey_ModAlt` and `ImGuiKey_ModSuper`.

Maybe you can have a `UnrealToImGuiModKeyMap` map `UnrealToImGuiKeyMap` just like as:
```
{
    EKeys::LeftShift : ImGuiKey_ModShift,
    EKeys::RigthShift : ImGuiKey_ModShift,
    EKeys::LeftControl : ImGuiKey_ModCtrl,
    EKeys::RightControl : ImGuiKey_ModCtrl,
    EKeys::LeftAlt : ImGuiKey_ModAlt,
    EKeys::RightAlt : ImGuiKey_ModAlt,
    EKeys::LeftCommand : ImGuiKey_ModSuper,
    EKeys::RightCommand : ImGuiKey_ModSuper,
}
```

And then:
```
if(ImGuiInterops::UnrealToImGuiModKey(Key) != ImGuiKey_None)
{
    // Will need UnrealToImGuiModKey function just like UnrealToImGuiKey, maybe without the log
    IO.AddKeyEvent(ImGuiInterops::UnrealToImGuiModKey(Key), bIsDown);
}
```

Reply:
Without these anything that uses Mod keys break with this PR change